### DISCUSSION TITLE
Additional Interferograms Alignment Issue

### DISCUSSION THREAD

**Discussion by krkrueger** [gmtsar/gmtsar#544] [Q&A]
Hey everyone,

Just ran into a slight issue that I was hoping to be able to get a little more understanding on.  I have created a bunch of interferograms over an area, of about 10 months.  When I did the 10 months' worth of intfs everything looked to align well. 2 new Sentinel 1 images became available, so in order to avoid rerunning everything, I looked to modify the workflow to only include, the newest 2 intf. My intf.in only includes the 4 new intf that are created from the 2 new images.

It looks like all the new intf products, corr_ll.grd, los_ll.grd, are all slightly offset (like 100m off) from the last 10 months of intf.  I am relatively new to the intf process, so I am sure I am missing a step when I calculate the new intfs. I would guess it is a process in the creation of the trans.dat (as this has caused me issues before). 

My process is:
1. to recalculate the SLC data for everything old and new combined.
2. I create the intf without unwrap or geocode for just the new 4 intf for F1 and F2.
3. I then run the merge for just the 4 new intf
4. Unwrap and geocode 4 new intf

Im sure there is a step somewhere I am missing that would allow for you to continue on adding new intf in the same project without having to create a complete new one, but not exactly sure where to start for such a minor issue.

My one thought was the merge file maybe during the merge step.

I use this little command to grab all the new files from F2 that do not exist in my merge folder.

comm -23 <(ls ../F2/intf_all) <(ls intf) > intf.list
create_merge_input.csh intf.list .. $2 > merge.list

I am attaching the original merge file, and the merge file when only selecting the new images.
[merge - new.txt](https://github.com/gmtsar/gmtsar/files/9975789/merge.-.new.txt)

 
[merge - old.txt](https://github.com/gmtsar/gmtsar/files/9975795/merge.-.old.txt)

**Comment by Xiaohua-Eric-Xu** [comment]
For merge_batch.csh, you need to set the first line of your input, a pair that contains the supermaster image (the one you aligned to) as the reference image (i.e. in aaaaaaa_bbbbbbb, aaaaaaa is the ref image). Also, you need to make sure your alignment are done selecting the same supermaster image.

**Reply by krkrueger** [reply]
Is there a way to use the create_merge_input to do this? It creates the merge.list for you, how do you tell it what your master image is? Because even in the old version, the master is not at the top.  Also, I already merged the master image, are you saying that I need to merge that one again in the second round? I will just pull 2 lines from the merge list that is doing 2 new intf.

../F1/intf_all/2022271_2022295/:S1_20220929_ALL_F1.PRM:S1_20221023_ALL_F1.PRM,../F2/intf_all/2022271_2022295/:S1_20220929_ALL_F2.PRM:S1_20221023_ALL_F2.PRM
../F1/intf_all/2022283_2022295/:S1_20221011_ALL_F1.PRM:S1_20221023_ALL_F1.PRM,../F2/intf_all/2022283_2022295/:S1_20221011_ALL_F2.PRM:S1_20221023_ALL_F2.PRM

My master image is one from 20220613. So are you saying I need to add a line to the merge like this, that will redo the master image again:

../F1/intf_all/2022163_2022175/:S1_20220613_ALL_F1.PRM:S1_20220625_ALL_F1.PRM,../F2/intf_all/2022163_2022175/:S1_20220613_ALL_F2.PRM:S1_20220625_ALL_F2.PRM

And then alignment done selecting the same super image, does that mean that the first .PRM file in all lines should be S1_20220613_ALL_F1.PRM?

So my merge.list should look like:

../F1/intf_all/2022163_2022175/:S1_20220613_ALL_F1.PRM:S1_20220625_ALL_F1.PRM,../F2/intf_all/2022163_2022175/:S1_20220613_ALL_F2.PRM:S1_20220625_ALL_F2.PRM
../F1/intf_all/2022271_2022295/:S1_20220613_ALL_F1.PRM:S1_20221023_ALL_F1.PRM,../F2/intf_all/2022271_2022295/:S1_20220613_ALL_F2.PRM:S1_20221023_ALL_F2.PRM
../F1/intf_all/2022283_2022295/:S1_20220613_ALL_F1.PRM:S1_20221023_ALL_F1.PRM,../F2/intf_all/2022283_2022295/:S1_20220613_ALL_F2.PRM:S1_20221023_ALL_F2.PRM

Any way to input the master to get the create_merge_input.csh to do this?

Thanks for the help!

**Reply by Xiaohua-Eric-Xu** [reply]
That's a good idea, I'll add an additional parameter for that. Not right now though.

**Reply by krkrueger** [reply]
Sorry, I went on vacation for a few weeks! Thanks for the responses, I am assuming the format I have exampled in my second response would be the correct one for the merge file assuming there is a super master? I think I can make some changes to the script on my end to get this set up if that is the case.

Thanks again!

**Reply by Xiaohua-Eric-Xu** [reply]
Sorry I haven't got time to work on this. If you are able to get it working, let me know and I can help put it to the GMTSAR distribution.
